doc/Command_Index_User.tex

```diff
@@ -1408,6 +1408,23 @@ \subsubsection{Ancient Notation}
   \#2 & integer & point size at which the font should be loaded\\
 \end{argtable}
 
+\subsubsection{Alterations (Flats, Naturals, and Sharps)}\label{setalterationeffect}
+
+\macroname{\textbackslash gresetalterationeffect}{\{\#1\}}{gregoriotex-signs.tex}
+Macro to set the effect of alterations (flats and sharps), for the purpose of deciding whether to print soft alterations (see \S\ref{softalterations}).
+
+\begin{argtable}
+  \#1 & \texttt{note} (default) & effect ends immediately \\
+      & \texttt{line} & effect ends at the end of the line \\
+\end{argtable}
+
+Soft flats are printed if no previous flat (on the same pitch) is in effect. Similarly for soft sharps. Soft naturals are printed if a previous flat or sharp (on the same pitch) is in effect.
+
+Thus, under \verb|\gresetalterationeffect{note}|, all sharps and flats lose their effect immediately, so all soft sharps and flats are printed, but no soft naturals are printed.
+Under \verb|\gresetalterationeffect{line}|, all sharps and flats are in effect until the end of the line (unless cancelled by a natural), which is the convention used in Dominican chant scores.
+
+(Argument \#1 is actually a comma-separated list of keywords, but currently, it must not be empty, and \texttt{note,line} has the same effect as \texttt{note}. In the future, other options like \texttt{word} and \texttt{bar} may be added.)
+
```

Comment:
The most confusing thing for me is I cannot easily tell if I need to use this macro or not. Perhaps a quick note about the default behavior of gregoriotex could be added. (cf. note on changelog)

Also why is `note, line` equal to `note`? Does the first or last option have priority? Does this macro need multiple options?

Reply:
I'll make these changes, but also have a question. I'm doubtful that `\grealterationeffect{note}` has any real use. It just seemed like the most neutral thing to do. Would it be preferable to instead make `line`, which is actually useful, the default? This would not break any existing code, because it only affects soft alterations, which are not used in existing code.

Reply:
I think making soft alterations default to `line` is fine.  As you say, this doesn't break existing scores because hard alterations aren't affected and were the only thing that existed previously.

Along those line, however, a note in UPGRADE about how to modify scores to make use of the new feature might be in order.